packages/compartment-mapper/src/link.js

```diff
@@ -356,9 +357,33 @@ export const link = (
 
   const pendingJobs = [];
 
+  /** @type {Record<string, Language>} */
+  const customLanguageForExtension = Object.create(null);
+  for (const { parser, extensions, language } of parsers) {
+    if (
+      language in parserForLanguage &&
+      parserForLanguage[language] !== parser
+    ) {
+      throw new Error(`Parser for language ${q(language)} already defined`);
+    }
+    parserForLanguage[language] = parser;
+    for (const extension of extensions) {
+      if (
+        extension in customLanguageForExtension &&
+        customLanguageForExtension[extension] !== language
+      ) {
+        throw new Error(
+          `Extension ${q(extension)} already assigned language ${q(customLanguageForExtension[extension])}`,
+        );
+      }
+      customLanguageForExtension[extension] = language;
+    }
+  }
```

Comment:
I’m beginning to suspect it’s better to thread separate “global” options for `parserForLanguage` and `languageForExtension` than to merge them into `parsers` and then unmerge them here. In https://github.com/endojs/endo/pull/2294, I would have to make `@endo/compartment-mapper/import-parsers.js` express an opinion about the language for each extension, and there is no clear opinion at that juncture.

Reply:
This branch captures my investigation and I am convinced that threading `parserForLanguage` and `languageForExtension` as separate options (instead of consolidated custom `parsers`) is robust with less ceremony and will compose better with changes I have coming for #400 

https://github.com/endojs/endo/compare/boneskull/native-parser...kriskowal-native-parser